compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -7606,6 +7608,21 @@ private BType checkMappingField(RecordLiteralNode.RecordField field, BType mappi
                             }
                         }
                     }
+                    if (!spreadRecordType.sealed) {
+                         if (mappingRecordType.sealed) {
+                             dlog.error(spreadExpr.pos,
+                                     DiagnosticErrorCode.INVALID_SPREAD_FIELD_TO_CREATE_CLOSED_RECORD_FROM_OPEN_RECORD,
+                                     spreadRecordType);
```

Comment:
```ballerina
type Address record {
    int street;
};

public function main() {
    string address = "10";
    record {|int street; string...;|} _ = {street: 10, ...address};
}

```
When considering the above scenario, the error is 'incompatible types: expected a map or a record, found 'string'.
So, we can use a similar error in the following scenario as well.
```ballerina
type Address record {
    string street;
};

public function main() {
    Address address = {street: "Main Street", "number": 40};
    record {|string street;|} _ = {...address}; // no compilation error
}
```